### DISCUSSION TITLE
Display client IP

### DISCUSSION THREAD

**Discussion by eponymic** [zauberzeug/nicegui#875] [Q&A] [answered]
### Question

Using the example code in the documentation, I am trying to display the client IP address on the page.  I only every see localhost, 127.0.0.1, even when connecting from my phone.  In case it helps, I am running the code in PyCharm.  Should I see the connecting device's IP address (i.e. the local PC or phone), whatever device has the browser open pointing at the server?

```py
@ui.page('/')
    async def wait_for_connection(client: Client):
      with ui.header() as head:
        ....
      await client.connected()
      await asyncio.sleep(2)
      print(client.__dict__)
      ui.label(f'The IP address {client.ip} was obtained from the websocket.')
```

**Comment by rodja** [comment]
Oh wow. Thanks for spotting this bug. The line

https://github.com/zauberzeug/nicegui/blob/3f594348b70aa9e02eb462fb34406aa0e67348b1/nicegui/client.py#L56

seems to be just wrong.

**Comment by rodja** [comment, marked as answer]
I just created pull request #877 to fix this. Until next release you can use this:

```py
@ui.page('/')
async def index(client: Client):
    await client.connected()
    ip = client.environ['asgi.scope']['client'][0]
    ui.label(f'The IP address {ip} was obtained from the websocket.')
```

**Reply by eponymic** [reply]
This worked.  Thank you very much
